flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatFactory.java

```diff
@@ -19,103 +19,48 @@
-import org.apache.flink.api.common.serialization.DeserializationSchema;
-import org.apache.flink.api.common.serialization.SerializationSchema;
-import org.apache.flink.api.common.typeinfo.TypeInformation;
 import org.apache.flink.configuration.ConfigOption;
 import org.apache.flink.configuration.ReadableConfig;
-import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.connector.file.src.FileSourceSplit;
+import org.apache.flink.connector.file.src.impl.StreamFormatAdapter;
+import org.apache.flink.connector.file.src.reader.BulkFormat;
+import org.apache.flink.formats.common.Converter;
 import org.apache.flink.table.connector.ChangelogMode;
-import org.apache.flink.table.connector.format.DecodingFormat;
-import org.apache.flink.table.connector.format.EncodingFormat;
-import org.apache.flink.table.connector.sink.DynamicTableSink;
-import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.connector.Projection;
+import org.apache.flink.table.connector.format.BulkDecodingFormat;
+import org.apache.flink.table.connector.format.ProjectableDecodingFormat;
+import org.apache.flink.table.connector.source.DynamicTableSource.Context;
 import org.apache.flink.table.data.RowData;
-import org.apache.flink.table.factories.DeserializationFormatFactory;
+import org.apache.flink.table.factories.BulkReaderFormatFactory;
 import org.apache.flink.table.factories.DynamicTableFactory;
-import org.apache.flink.table.factories.FactoryUtil;
-import org.apache.flink.table.factories.SerializationFormatFactory;
 import org.apache.flink.table.types.DataType;
 import org.apache.flink.table.types.logical.RowType;
 
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.JsonNode;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.dataformat.csv.CsvSchema;
+
 import org.apache.commons.lang3.StringEscapeUtils;
 
 import java.util.Collections;
-import java.util.HashSet;
 import java.util.Set;
 
 import static org.apache.flink.formats.csv.CsvFormatOptions.ALLOW_COMMENTS;
 import static org.apache.flink.formats.csv.CsvFormatOptions.ARRAY_ELEMENT_DELIMITER;
 import static org.apache.flink.formats.csv.CsvFormatOptions.DISABLE_QUOTE_CHARACTER;
 import static org.apache.flink.formats.csv.CsvFormatOptions.ESCAPE_CHARACTER;
 import static org.apache.flink.formats.csv.CsvFormatOptions.FIELD_DELIMITER;
-import static org.apache.flink.formats.csv.CsvFormatOptions.IGNORE_PARSE_ERRORS;
 import static org.apache.flink.formats.csv.CsvFormatOptions.NULL_LITERAL;
 import static org.apache.flink.formats.csv.CsvFormatOptions.QUOTE_CHARACTER;
 
-/**
- * Format factory for providing configured instances of CSV to RowData {@link SerializationSchema}
- * and {@link DeserializationSchema}.
- */
+/** CSV format factory for file system. */
+//TODO: Rename back to CsvFileSystemFormatFactory? @Arvid
```

Comment:
Why do you think so?
Can we find a name that doesn't start with Csv so it pops up later in auto-complete? `TableCsv...` or `InternalCsv...`?

Reply:
I had this thought because `BulkReader` is a `flink-connector-file` interface, and `CsvBulkWriter` writer into `FSDataOutputStream`. Both the encoder and the decoder exposed by the factory are for files.
I like the `TableCsvFormatFactory`, renamed.

Reply:
I guess we could pass the `OutputStream` instead of the  `FSDataOutputStream`, but that would mean adding a check via `isinstanceof` in the `finish()` method for calling `sync()` for files.

Reply:
I like the `TableCsvFormatFactory` - will rename it.

Reply:
I forgot that the `BulkWriter` factory is passed an `FSDataOutputStream`, so technically both ins and outs of this factory are only for supporting files.